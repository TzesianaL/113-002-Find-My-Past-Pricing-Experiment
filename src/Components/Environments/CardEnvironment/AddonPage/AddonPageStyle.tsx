import styled from 'styled-components'
import { AddonGroupStyle } from '../AddonGroup/AddonGroupStyle'

export const AddonPageStyle = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: center;
  margin-bottom: 3em;
  background: #f2f2f2;
}

  .introText {
    font-size: 1.2em;
    max-width: 50em;
    @media screen and (max-width:600px) {
      max-width: 16em;
     
    }
  }

  .footerText {
    font-size: 1.2em;
    max-width: 50em;
    @media screen and (max-width: 600px) {
      max-width: 16em;
    }
  }
  h3,h2,h1{
    color: black;
  }

  ${AddonGroupStyle} {
    display: flex;
    flex-direction: column;
    // flex-wrap: wrap;
    background: #f2f2f2;
    justify-content: center;
  }
`
