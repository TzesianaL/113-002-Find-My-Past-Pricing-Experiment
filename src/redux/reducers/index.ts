import { combineReducers } from 'redux'
import { dataStoreReducer } from './dataStore/dataStore'
import { outputStoreReducer } from './output/outputSlice'
import { logicStoreReducer } from './logicStore/logicStore'
import { pageDurationStoreReducer } from './pageDurationStore/pageDurationStore'
import { actionLogStoreReducer } from './actionLogStore/actionLogStore'

const rootReducer = combineReducers({
  data: dataStoreReducer,
  logic: logicStoreReducer,
  output: outputStoreReducer,
  pageDuration: pageDurationStoreReducer,
  actionLog: actionLogStoreReducer,
})

export default rootReducer
export type RootState = ReturnType<typeof rootReducer>
