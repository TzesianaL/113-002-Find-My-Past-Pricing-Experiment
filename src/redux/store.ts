import { configureStore, EnhancedStore } from '@reduxjs/toolkit'
import merge from 'lodash-es/merge'
import rootReducer, { RootState } from './reducers'
import { initialState as dataInitialState } from './reducers/dataStore/dataStore'
import { initialState as actionLogInitialState } from './reducers/actionLogStore/actionLogStore'
import { initialState as logicInitialState } from './reducers/logicStore/logicStore'
import { initialState as outputInitialState } from './reducers/output/outputSlice'
import { initialState as pageDurationInitialState } from './reducers/pageDurationStore/pageDurationStore'

export const store = configureStore({
  reducer: rootReducer,
})

const defaultStoreState: RootState = {
  data: dataInitialState,
  logic: logicInitialState,
  actionLog: actionLogInitialState,
  output: outputInitialState,
  pageDuration: pageDurationInitialState,
}

// eslint-disable-next-line @typescript-eslint/ban-types
type DeepPartial<T> = T extends object
  ? {
      [P in keyof T]?: DeepPartial<T[P]>
    }
  : T

export const preloadedStore = (
  state: DeepPartial<RootState>
): EnhancedStore<RootState> =>
  configureStore({
    reducer: rootReducer,
    preloadedState: merge({}, defaultStoreState, state),
  })
