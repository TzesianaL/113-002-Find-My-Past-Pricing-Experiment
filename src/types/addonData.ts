import { Periods } from './periods'

export interface AddonData {
  id: string
  title: string
  subtitle?: string
  features?: string[]
  price?: number
  pricePeriod?: Periods
  includedProductIDs?: string[]
  excludedProductIDs?: string[]
  mutex?: boolean
  group?: number
  mutexGroup4?: boolean
  mutexGroup2?: boolean
}
