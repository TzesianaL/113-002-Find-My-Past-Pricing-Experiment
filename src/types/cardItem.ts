import { Periods } from './periods'

interface BasketCardOptions {
  prePriceText?: string
  postPriceText?: string
}

export type ContractDuration = '24 months' | '30 days'

// Type guard function for runtime validation of ContractDuration
export const isContractDuration = (
  value: string
): value is ContractDuration => {
  return value === '24 months' || value === '30 days'
}

export interface CardItem {
  id: string
  title: string
  subtitle: string
  subtitle2?: string
  price: number
  currency?: string
  priceString?: string
  features: string[]
  cardColor?: string
  highlightedText?: string
  headerImage?: string
  upfrontPrice?: number
  basketOptions?: BasketCardOptions
  pricePeriod?: Periods
  contractDuration?: ContractDuration
  parentSpeedId?: string
}

export interface SpeedCardItem {
  title: string
  subtitle: string
  price: number
  currency?: string
  highlightedText?: string
  id: string
  pricePeriod?: Periods
  purchaseButtonText?: string
  mainSpeed: string
  downloadSpeed: string
  downloadExample: string
  downloadTime: string
  uploadSpeed: string
  uploadExample: string
  uploadTime: string
  speed?: number
}
