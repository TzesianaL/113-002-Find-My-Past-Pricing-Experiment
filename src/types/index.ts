export * from './addonData'
export * from './cardItem'
export * from './layout'
export * from './periods'
export * from './reduxData'
export * from './splashItem'
export * from './pcw'
export * from './environmentType'
