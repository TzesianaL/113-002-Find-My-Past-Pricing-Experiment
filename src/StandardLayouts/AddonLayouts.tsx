import { BlockLayout, BlockEnums } from 'types'

export const horizontalThreeColumnAddonLayout: BlockLayout = {
  blockType: BlockEnums.HStack,
  blocks: [
    {
      blockType: BlockEnums.VStack,
      blocks: [
        { blockType: BlockEnums.Title },
        {
          blockType: BlockEnums.Subtitle,
          styleOverride: {
            color: 'gray',
          },
        },
      ],
      styleOverride: {
        flex: '0 0 20em',
        alignSelf: 'flex-start',
        alignItems: 'center',
      },
    },
    {
      blockType: BlockEnums.Features,
      styleOverride: {
        flex: '1 0 15em',
        maxWidth: '20em',
      },
    },
    {
      blockType: BlockEnums.Price,
      styleOverride: { flex: '0 0 auto', margin: '0 2em' },
    },
    {
      blockType: BlockEnums.AddRemoveButton,
      styleOverride: {
        flex: '0 0 7em',
        margin: '0 2em',
      },
    },
  ],
}

export const cardAddonLayout: BlockLayout = {
  blockType: BlockEnums.VStack,
  blocks: [
    {
      blockType: BlockEnums.Title,
      styleOverride: {
        color: 'black',
        alignSelf: 'left',
        textAlign: 'left',
      },
    },
    {
      blockType: BlockEnums.Subtitle,
      styleOverride: {
        color: 'black',
        alignSelf: 'left',
        textAlign: 'left',
      },
    },
    {
      blockType: BlockEnums.Features,
      styleOverride: {
        color: 'black',
      },
    },
    {
      blockType: BlockEnums.Price,
      styleOverride: {
        color: 'black',
        alignSelf: 'left',
        fontWeight: 'bold',
      },
    },
    {
      blockType: BlockEnums.AddRemoveButton,
      styleOverride: {
        alignSelf: 'center',
        width: '300px',
        background: 'black',
        color: 'white',
      },
    },
  ],
  styleOverride: {
    width: '18em',
    height: '15em',
    border: '2px solid black',
    padding: '1em',
    background: 'white',
    justifyContent: 'space-around',
    margin: '1em',
    color: 'black',
    borderRadius: '5px',
  },
}

export const WifiMaxAddonLayout: BlockLayout = {
  blockType: BlockEnums.VStack,
  blocks: [
    {
      blockType: BlockEnums.Title,
      styleOverride: {
        color: 'black',
        alignSelf: 'left',
        textAlign: 'left',
      },
    },
    {
      blockType: BlockEnums.Subtitle,
      styleOverride: {
        color: 'black',
        alignSelf: 'left',
        textAlign: 'left',
      },
    },
    {
      blockType: BlockEnums.Features,
      styleOverride: {
        color: 'black',
      },
    },
    {
      blockType: BlockEnums.Price,
      styleOverride: {
        color: 'black',
        alignSelf: 'left',
        fontWeight: 'bold',
      },
    },
    {
      blockType: BlockEnums.AddRemoveButton,
      styleOverride: {
        alignSelf: 'center',
        width: '300px',
        background: 'black',
        color: 'white',
      },
    },
  ],
  styleOverride: {
    width: '18em',
    height: '16em',
    border: '2px solid black',
    padding: '1em',
    background: 'white',
    justifyContent: 'space-around',
    margin: '1em',
    color: 'black',
    borderRadius: '5px',
  },
}
export const highlightedCardAddonLayout: BlockLayout = {
  blockType: BlockEnums.VStack,
  blocks: [
    {
      blockType: BlockEnums.Title,
      styleOverride: {
        color: 'black',
      },
    },
    {
      blockType: BlockEnums.Subtitle,
      styleOverride: {
        color: 'black',
        alignSelf: 'center',
      },
    },
    {
      blockType: BlockEnums.Features,
      styleOverride: {
        color: 'black',
      },
    },
    {
      blockType: BlockEnums.Price,
      styleOverride: {
        color: 'black',
        alignSelf: 'center',
        fontWeight: 'bold',
      },
    },
    {
      blockType: BlockEnums.AddRemoveButton,
      styleOverride: {
        alignSelf: 'center',
        width: '200px',
        background: 'black',
        color: 'white',
      },
    },
  ],
  styleOverride: {
    width: '18em',
    height: '20em',
    border: '2px solid black',
    padding: '1em',
    background: 'white',
    justifyContent: 'space-around',
    margin: '1em',
    color: 'black',
    borderRadius: '5px',
  },
}
export const testLayout: BlockLayout = {
  blockType: BlockEnums.HStack,
  blocks: [
    {
      blockType: BlockEnums.VStack,
      blocks: [
        {
          blockType: BlockEnums.Title,
          styleOverride: {
            color: 'red',
          },
        },
        {
          blockType: BlockEnums.Subtitle,
          styleOverride: {
            color: 'gray',
          },
        },
      ],
      styleOverride: {
        flex: '0 0 50%',
        alignItems: 'center',
      },
    },
    {
      blockType: BlockEnums.Features,
      styleOverride: {
        flex: '1 0 auto',
        width: 'fit-content',
        alignItems: 'flex-start',
      },
    },
  ],
}

export const StorePageAddonLayout: BlockLayout = {
  blockType: BlockEnums.VStack,
  blocks: [
    {
      blockType: BlockEnums.Title,
      styleOverride: {
        color: 'black',
      },
    },
    {
      blockType: BlockEnums.Subtitle,
      styleOverride: {
        color: 'black',
        alignSelf: 'center',
      },
    },
    {
      blockType: BlockEnums.Features,
      styleOverride: {
        color: 'black',
      },
    },
    {
      blockType: BlockEnums.Price,
      styleOverride: {
        color: 'black',
        alignSelf: 'center',
        fontWeight: 'bold',
      },
    },
    {
      blockType: BlockEnums.AddRemoveButton,
      styleOverride: {
        alignSelf: 'center',
        width: '200px',
        background: 'black',
        color: 'white',
      },
    },
  ],
  styleOverride: {
    width: '20em',
    height: '28em',
    border: '2px solid black',
    padding: '1em',
    background: 'white',
    justifyContent: 'space-around',
    margin: '1em',
    color: 'black',
    borderRadius: '7px',
  },
}

export const StorePageAddonLayoutScenarios123: BlockLayout = {
  blockType: BlockEnums.VStack,
  blocks: [
    {
      blockType: BlockEnums.Title,
      styleOverride: {
        color: 'black',
      },
    },
    {
      blockType: BlockEnums.Subtitle,
      styleOverride: {
        color: 'black',
        alignSelf: 'center',
        textAlign: 'center',
      },
    },
    {
      blockType: BlockEnums.Features,
      styleOverride: {
        color: 'black',
        textAlign: 'center',
      },
    },
    {
      blockType: BlockEnums.Price,
      styleOverride: {
        color: 'black',
        alignSelf: 'center',
        fontWeight: 'bold',
      },
    },
    {
      blockType: BlockEnums.AddRemoveButton,
      styleOverride: {
        alignSelf: 'center',
        width: '200px',
        background: 'black',
        color: 'white',
      },
    },
  ],
  styleOverride: {
    width: '20em',
    height: '31em',
    border: '2px solid #83358e',
    padding: '1em',
    background: 'white',
    justifyContent: 'space-around',
    margin: '1em',
    color: 'black',
    borderRadius: '7px',
  },
}

export const xtraBroadBandLayout: BlockLayout = {
  blockType: BlockEnums.VStack,
  blocks: [
    {
      blockType: BlockEnums.Title,
      styleOverride: {
        color: 'black',
      },
    },
    {
      blockType: BlockEnums.Subtitle,
      styleOverride: {
        color: 'black',
        alignSelf: 'center',
        textAlign: 'center',
      },
    },
    {
      blockType: BlockEnums.Features,
      styleOverride: {
        color: 'black',
        textAlign: 'center',
      },
    },
    {
      blockType: BlockEnums.Price,
      styleOverride: {
        color: 'black',
        alignSelf: 'center',
        fontWeight: 'bold',
      },
    },
    {
      blockType: BlockEnums.AddRemoveButton,
      styleOverride: {
        alignSelf: 'center',
        width: '200px',
        background: 'black',
        color: 'white',
      },
    },
  ],
  styleOverride: {
    width: '18em',
    height: '25em',
    border: '2px solid #228B22',
    padding: '1em',
    background: '#D3F8D3',
    justifyContent: 'space-around',
    margin: '1em',
    color: 'black',
    borderRadius: '7px',
  },
}
