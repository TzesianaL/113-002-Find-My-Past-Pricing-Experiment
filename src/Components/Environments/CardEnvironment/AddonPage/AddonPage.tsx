import React, { FC } from 'react'
import Button from 'Components/Shared/Button/Button'
import { BlockLayout, AddonData } from 'types'
import { useSelector } from 'react-redux'
import { useScrollToTop } from 'hooks/useScrollToTop'
import { WifiMaxAddonLayout } from 'StandardLayouts/AddonLayouts'
import {
  getAddonsInBasket,
  getBasketItem,
  getBasketSpeedItem,
  getCost,
  getSelectedCondition,
} from 'redux/reducers/logicStore/logicStore.selectors'
import AddonGroup from '../AddonGroup/AddonGroup'
import StoreBanner from '../../../Shared/StoreBanner/StoreBanner'
import { AddonPageStyle } from './AddonPageStyle'

export interface AddonPageProps {
  data: AddonData[]
  layout: BlockLayout
  bannerName?: string
  bannerLogoSrc?: string
  introText?: string
  footerText?: string
  onContinue?: () => void
}

/**
 * AddonPage is used to display the list of add ons, the banner with store  logo and continue button as well.
 * ThemeProvider component is used. By using the theme object we provide same styling options to all components.
 */
const AddonPage: FC<AddonPageProps> = ({
  data,
  layout,
  bannerName,
  bannerLogoSrc,
  introText,
  footerText,
  onContinue,
}) => {
  const currentPrice = useSelector(getCost)
  const selectedCondition = useSelector(getSelectedCondition)
  useScrollToTop()
  const selectedItem = useSelector(getBasketItem)

  const normalAddons = data.filter((addon) => addon.group === 3)
  const wifiMaxAddon = data.find((addon) => addon.group === 4)
  return (
    <AddonPageStyle>
      <div style={{ alignItems: 'left', width: '100%' }}>
        <StoreBanner name={bannerName} logoSrc={bannerLogoSrc} />
      </div>

      {introText && (
        <div
          className="introText"
          dangerouslySetInnerHTML={{ __html: introText }}
        />
      )}
      <h1>
        Add-ons
        <span style={{ fontWeight: 'normal' }}> (optional)</span>
      </h1>
      <AddonGroup data={normalAddons} layout={layout} />
      {console.log(
        'selectedCondition',
        typeof selectedCondition,
        selectedCondition
      )}
      {selectedCondition === 2 && (
        <>
          <h1>
            Add WIFI Max{' '}
            <span style={{ fontWeight: 'normal' }}>to your broadband</span>
          </h1>
          <AddonGroup
            data={wifiMaxAddon ? [wifiMaxAddon] : []}
            layout={WifiMaxAddonLayout}
          />
        </>
      )}
      {footerText && (
        <div
          className="footerText"
          dangerouslySetInnerHTML={{ __html: footerText }}
        />
      )}
      {onContinue && <Button onClick={onContinue}>Continue to basket </Button>}
      <br />
      <br />
    </AddonPageStyle>
  )
}

export default AddonPage
