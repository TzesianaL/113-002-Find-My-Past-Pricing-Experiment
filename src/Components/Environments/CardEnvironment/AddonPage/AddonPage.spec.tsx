import React from 'react'
import 'jest-styled-components'
import { render } from '@testing-library/react'
import { horizontalThreeColumnAddonLayout } from 'StandardLayouts/AddonLayouts'
import { AddonData } from 'types'
import { Provider } from 'react-redux'
import { store } from 'redux/store'
import AddonPage from './AddonPage'

describe('<AddonPage/>', () => {
  const Addons: AddonData[] = [
    {
      id: 'Addon_1',
      title: 'Basic',
      price: 9.99,
      features: ['Feature 1'],
    },
    {
      id: 'Addon_2',
      title: 'Better',
      price: 29.99,
      features: ['Feature 1', 'Feature 2'],
    },
    {
      id: 'Addon_3',
      title: 'Premium',
      price: 39.99,
      features: ['Feature 1', 'Feature 2', 'Feature 3', 'Feature 4'],
    },
  ]

  it('should render AddonPage component correctly', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <AddonPage
          data={Addons}
          layout={horizontalThreeColumnAddonLayout}
          bannerName="Test"
        />
      </Provider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
