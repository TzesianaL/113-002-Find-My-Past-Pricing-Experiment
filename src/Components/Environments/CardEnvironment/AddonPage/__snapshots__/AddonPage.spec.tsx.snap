// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddonPage/> should render AddonPage component correctly 1`] = `
<DocumentFragment>
  .c10 {
  color: #ffffff;
  background-color: #003D59;
  cursor: pointer;
  font-size: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0px;
  border-radius: 6px;
  padding: 0.7em 1em;
}

.c10:active {
  background-color: #336D89;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 img {
  margin: 1rem;
  width: 160px;
  border-radius: 10px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c8 {
  color: black;
  font-size: em;
  padding-left: 2em;
  padding-right: 2em;
  width: calc(100% - 4em);
}

.c8 li {
  padding-left: 0.25em;
  margin-bottom: 1.5em;
}

.c8 li::marker {
  content: '✓';
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  background-color: #6300AA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fill-available;
  width: -moz-available;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white !important;
  padding: 0.6rem 0 0.6rem 0;
}

.c1 p {
  width: 100%;
  font-size: rem;
}

.c1 img {
  padding: 0 1em;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1.2rem;
  margin: 0em 0em 0.5em 0em;
}

.c9 del {
  color: red;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 3em;
  background: #333333;
}

.c0 .introText {
  font-size: 1.2em;
  max-width: 50em;
}

.c0 .footerText {
  font-size: 1.2em;
  max-width: 50em;
}

.c0 .c2 {
  margin-top: 1em;
  margin-bottom: 5em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  background: ##333333;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <p>
        Test
      </p>
    </div>
    <div
      class="c2 c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
            style="flex: 0 0 20em; align-self: flex-start; align-items: center;"
          >
            <div
              class="c7"
            >
              Basic
            </div>
          </div>
          <ul
            class="c8"
            style="flex: 1 0 15em; max-width: 20em;"
          >
            Feature 1
          </ul>
          <div
            class="c9"
            style="flex: 0 0 auto; margin: 0px 2em;"
          >
            £9.99
          </div>
          <button
            class="c10"
            style="flex: 0 0 7em; margin: 0px 2em;"
          >
            Add
          </button>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
            style="flex: 0 0 20em; align-self: flex-start; align-items: center;"
          >
            <div
              class="c7"
            >
              Better
            </div>
          </div>
          <ul
            class="c8"
            style="flex: 1 0 15em; max-width: 20em;"
          >
            Feature 1
          </ul>
          <div
            class="c9"
            style="flex: 0 0 auto; margin: 0px 2em;"
          >
            £29.99
          </div>
          <button
            class="c10"
            style="flex: 0 0 7em; margin: 0px 2em;"
          >
            Add
          </button>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <div
            class="c6"
            style="flex: 0 0 20em; align-self: flex-start; align-items: center;"
          >
            <div
              class="c7"
            >
              Premium
            </div>
          </div>
          <ul
            class="c8"
            style="flex: 1 0 15em; max-width: 20em;"
          >
            Feature 1
          </ul>
          <div
            class="c9"
            style="flex: 0 0 auto; margin: 0px 2em;"
          >
            £39.99
          </div>
          <button
            class="c10"
            style="flex: 0 0 7em; margin: 0px 2em;"
          >
            Add
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
